code/game/objects/machinery/vending/vending.dm

```diff
@@ -508,7 +512,9 @@ GLOBAL_LIST_INIT(vending_white_items, typecacheof(list(
 			src.speak(src.vend_reply)
 			src.last_reply = world.time
 
-	release_item(R, vend_delay)
+	var/obj/item/new_item = release_item(R, vend_delay)
+	if(istype(new_item))
+		user.put_in_any_hand_if_possible(new_item, warning = FALSE)
```

Comment:
Internally this called equip to slot, which seems to apply armor from the item to the mob.
This caused some issues where marines were stacking double armor and becoming invulnerable.

Im not sure exactly why this is happening here and not elsewhere, but if vend essential, heavy helmet etc, and check your chest limb you'll see around 65 armor before you equip any gear.

Reply:
Sounds like a problem with armor code not with this